tests/Doctrine/Tests/ORM/Functional/Ticket/GH7505Test.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Annotation as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH7505Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH7505AbstractResponse::class,
+            GH7505ArrayResponse::class,
+            GH7505TextResponse::class,
+        ]);
+    }
+
+    public function testSimpleArrayTypeHydratedCorrectly() : void
+    {
+        $arrayResponse = new GH7505ArrayResponse();
+        $this->em->persist($arrayResponse);
+
+        $textResponse = new GH7505TextResponse();
+        $this->em->persist($textResponse);
+
+        $this->em->flush();
+        $this->em->clear();
+
+        $repository = $this->em->getRepository(GH7505AbstractResponse::class);
+
+        /** @var GH7505ArrayResponse $arrayResponse */
+        $arrayResponse = $repository->find($arrayResponse->id);
+        static::assertEquals($arrayResponse->value, []);
```

Comment:
The order of arguments here are wrong, the expectation comes first. Please also use `self` instead of `static` (the class is final and won't never be inherited). Lastly, `assertSame()` would be better for this case than `assertEquals()`.